tokio/src/io/util/async_read_ext.rs

```diff
@@ -105,8 +105,10 @@ cfg_io_util! {
         /// async fn read(&mut self, buf: &mut [u8]) -> io::Result<usize>;
         /// ```
         ///
-        /// This function does not provide any guarantees about whether it
-        /// completes immediately or asynchronously
+        /// This method does not provide any guarantees about whether it
```

Comment:
I would probably make this its own section.

```
# Cancel safety

This method is cancel-safe and can be used safely [...]
```

The bit about not providing guarantees about whether it completes immediately or not is good as well, but I am imaging users want to quickly check if the method is cancel-safe or not and we want to make the doc easy to skim for that data.